modules/local/functions.nf

```diff
@@ -109,30 +109,26 @@ def extract_bam(tsvFile) {
 // Create a channel of germline FASTQs from a directory pattern: "my_samples/*/"
 // All FASTQ files in subdirectories are collected and emitted;
 // they must have _R1_ and _R2_ in their names.
-def extract_fastq_from_dir(pattern) {
-    def fastq = Channel.create()
-    // a temporary channel does all the work
-    Channel
-        .fromPath(pattern, type: 'dir')
-        .ifEmpty { error "No directories found matching pattern '${pattern}'" }
-        .subscribe onNext: { sampleDir ->
-            // the last name of the sampleDir is assumed to be a unique sample id
-            sampleId = sampleDir.getFileName().toString()
+// All FASTQ files are assumed to be from the same sample.
+def extract_fastq_from_dir(folder) {
+    sample = file(folder).getFileName().toString()
+
+    fastq = Channel.fromFilePairs(folder + '/*{_R1_,_R2_}*.fastq.gz')
+        .ifEmpty { error "No directories found matching folder '${folder}'" }
+
+    fastq = fastq.map{ run, pair ->
+        def meta = [:]
+        meta.patient = sample
+        meta.sample  = meta.patient
+        meta.gender  = 'ZZ' // unused
```

Comment:
Why is this unused? In the input.tsv this is required and relevant for ASCAT, no?

Reply:
Ah ok, this for running sarek on fasts directly without a tsv? and then not all the analysis can be performed or how does it work?

Reply:
That's exactly it.
When using --input on a fastq directory, it's assumed that all fastq pairs are from the same normal sample.
Then gender is not used, since it's only relevant for CNV.
And status is 0

Reply:
nice, didn't realise sarek has this functionality as well :)